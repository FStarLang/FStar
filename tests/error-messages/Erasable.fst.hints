[
  ".Jw=Ä¡òA\u001b\u0011Ì\u0005˜ÓN\u001b",
  [
    [
      "Erasable.__proj__This__item___0",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_3c192a2edf78cec4f1eb553cb69a4e8f"
      ],
      0,
      "47004d7e7f21c61d548295a43a746728"
    ],
    [
      "Erasable.__proj__That__item___0",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_ec597d6979f8847712206596e24a9e2d"
      ],
      0,
      "31153b7986378cd915d0beb9dbf62bcd"
    ],
    [
      "Erasable.test",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "disc_equation_Erasable.That",
        "disc_equation_Erasable.This", "fuel_guarded_inversion_Erasable.t",
        "projection_inverse_BoxBool_proj_0"
      ],
      0,
      "569524b3d6c0e1e16c5d0e9e707f5423"
    ],
    [
      "Erasable.test_promotion",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "disc_equation_Erasable.That",
        "disc_equation_Erasable.This", "fuel_guarded_inversion_Erasable.t",
        "projection_inverse_BoxBool_proj_0"
      ],
      0,
      "183931908526447b5fc9590bc5e4481f"
    ]
  ]
]