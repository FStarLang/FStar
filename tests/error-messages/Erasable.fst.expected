>> Got issues: [
Erasable.fst(6,0-8,17): (Error 162) Incompatible attributes and qualifiers: erasable types do not support decidable equality and must be marked `noeq`
>>]
>> Got issues: [
Erasable.fst(18,2-20,15): (Error 34) Computed type "Prims.int" and effect "GHOST" is not compatible with the annotated type "Prims.int" effect "Tot"
>>]
>> Got issues: [
Erasable.fst(28,42-28,52): (Error 34) Computed type "Prims.int" and effect "GTot" is not compatible with the annotated type "Prims.int" effect "Tot"
>>]
>> Got issues: [
Erasable.fst(41,12-41,15): (Error 162) Illegal attribute: the `erasable` attribute is only permitted on inductive type definitions and abbreviations for non-informative types. Prims.nat is considered informative.
>>]
>> Got issues: [
Erasable.fst(52,0-52,17): (Error 162) Mismatch of attributes between declaration and definition: Declaration is marked `erasable` but the definition is not
>>]
>> Got issues: [
Erasable.fst(59,0-59,29): (Error 162) Mismatch of attributes between declaration and definition: Declaration is marked `erasable` but the definition is not
>>]
Erasable.fst(49,4-49,11): (Warning 240) Erasable.e_nat_2 is declared but no definition was found; add an 'assume' if this is intentional
Erasable.fst(56,4-56,11): (Warning 240) Erasable.e_nat_3 is declared but no definition was found; add an 'assume' if this is intentional
Verified module: Erasable
All verification conditions discharged successfully
